formatter_test.go

```diff
@@ -93,3 +93,524 @@ func TestFormatDiff(t *testing.T) {
 	checkOK(map[string]interface{}{"a": 1}, map[string]interface{}{})
 	checkOK([]interface{}{"a"}, []interface{}{})
 }
+
+func TestFormatDataFailureActual(t *testing.T) {
```

Comment:
ok will implement the table driven one

Reply:
ok, will add test for AssertValid too

Reply:
btw perhaps I will solve the comment part by part

would make it done by this week

i will let you know if this PR ready to review again ya

Reply:
I have done for the `TestFormatDataFailureActual`

https://github.com/gavv/httpexpect/pull/202/commits/0339d6b222eaa3ada0c2e580f78b575120d7be03

LMK if you have comments about it

thanks

Reply:
The new test looks good.

It seems that assertFn is very similar for all cases. Maybe replace it with data too? Instead of assertFn, add fields HaveActual and FormattedActual. This way the test would be fully data driven. What do you think?

Reply:
@gavv I see. Agree. I think making it data driven is also more readable and straight forward

Will do the update